app/src/main/java/com/example/musicplayer/features/features/PlayedMusic/PlayedViewModel.kt

```diff
@@ -14,30 +14,26 @@ class PlayedViewModel(private val playedRepository: PlayedRepository) : ViewMode
     private var _time: MutableStateFlow<Time> = MutableStateFlow(Time())
     val time: StateFlow<Time> = _time
 
-    private val handler = Handler()
-    private lateinit var runnable: Runnable
     private val interval = 1000L
 
     fun getContentUri(appendedId: Long): Uri =
-    playedRepository.getUriId(appendedId)
+        playedRepository.getUriId(appendedId)
 
-    fun initializeWatcher() {
-        runnable = object : Runnable {
-            override fun run() {
-                getMusicCurrentPosition()
-                handler.postDelayed(this, interval)
-            }
-        }
-        handler.postDelayed(runnable, interval)
-    }
+    fun getConnection() = playedRepository.getConnection()
 
-    fun getMusicCurrentPosition() {
+    fun watchPlayBack() {
         viewModelScope.launch {
-            playedRepository.getPlayer().collect {
-                _time.value = it
+            delay(2000L)
```

Comment:
currentPosition and duration are the same for unknown reasons,therefore it is paused so the values match our conditions.